manager/orchestrator/update/updater.go

```diff
@@ -520,10 +521,41 @@ func (u *Updater) removeOldTasks(ctx context.Context, batch *store.Batch, remove
 	return removedTask, nil
 }
 
+// Checks if the current assigned node matches the Placement.Constraints
+// specified in the task spec for Updater.newService.
+func (u *Updater) nodeMatches(t *api.Task) bool {
+	if t.Spec.Placement == nil {
+		return true
+	}
+
+	constraints, _ := constraint.Parse(u.newService.Spec.Task.Placement.Constraints)
+
+	match := false
+	u.store.Update(func(tx store.Tx) error {
+		n := store.GetNode(tx, t.NodeID)
+		match = constraint.NodeMatches(constraints, n)
+		return nil
+	})
+
+	return match
+}
+
 func (u *Updater) isTaskDirty(t *api.Task) bool {
+	// Task is not dirty if:
+	// 1. Either the task spec didn't change at all, or
+	// 2. The placement constraints alone changed and the node currently assigned
+	// can satisfy the changed constraints.
+	if u.isTaskPlacementConstraintsOnlyDirty(t) && u.nodeMatches(t) {
+		return false
+	}
```

Comment:
This should go inside `orchestrator.IsTaskDirty`. Otherwise, when the orchestrators reconcile services, they will replace tasks that were intentionally not updated here.

Reply:
Thx for pointing this out. Done in next patch!